state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -472,6 +532,26 @@ pub mod pallet {
 					}
 				}
 			}
+			let scheduled_limit_orders = ScheduledLimitOrders::<T>::take(current_block);
+			for order_update in scheduled_limit_orders {
+				if let Err(err) = order_update
+					.clone()
+					.call
```

Comment:
I think we might want to wrap the dispatch in a storage transaction to ensure we roll back state if it fails.